libs/commons/Uri_.ml

```diff
@@ -26,3 +26,13 @@
 let of_string_opt (str : string) : Uri.t option =
   let uri = Uri.of_string str in
   if Uri.equal uri Uri.empty then None else Some uri
+
+
+let url_regex = Pcre2_.regexp "^https?://"
+let is_url config_path =
+  match Pcre2_.pmatch ~rex:url_regex config_path with
+  | Ok true -> true
+  | Ok false -> false 
+  | Error err ->
+      Log.warn (fun m -> m "Error Validation URL: %a" pp_error err);
+      false
```

Comment:
Here you can just have : 

```ocaml 
| Ok res -> res 
| Error err  -> (* your code *)
```

Or even better, just use:

```ocaml
Pcre2_.pmatch_noerr
```

that will do the logging for you, ie: 

```ocaml
let is_url config_path = Pcre2_.pmatch_noerr ~rex:url_regex config_pat in ...
```

(no need for `match` in that case).

Reply:
In fact your version will probably not compile, but my proposed solution with `pmatch_noerr` will.

Reply:
Notice that in `Pcre2_.ml`, line 27, we define: `module Log = (val Logs.src_log src : Logs.LOG)`.

This is used in the body of `pmatch_noerr` within `Pcre2_.ml`, but it's not visible in `Uri_.ml` which is why compilation failed.